Source/WebCore/page/Frame.cpp

```diff
@@ -73,7 +73,8 @@ bool Frame::isRootFrame() const
 
 void Frame::resetWindowProxy()
 {
-    m_windowProxy->detachFromFrame();
+    if (m_windowProxy->frame() == this)
```

Comment:
I feel like we ought to just add ASSERT(m_windowProxy->frame() == this) here, otherwise we've already done something wrong.